nexus/serializer.go

```diff
@@ -226,3 +240,22 @@ type compositeSerializer struct {
 var defaultSerializer Serializer = compositeSerializer{
 	serializerChain([]Serializer{nilSerializer{}, byteSliceSerializer{}, jsonSerializer{}}),
 }
+
+type errorMessageFailureConverter struct{}
+
+// ErrorToFailure implements FailureConverter.
+func (e errorMessageFailureConverter) ErrorToFailure(err error) Failure {
+	if err == nil {
+		return Failure{}
+	}
+	return Failure{
+		Message: err.Error(),
+	}
+}
+
+// FailureToError implements FailureConverter.
+func (e errorMessageFailureConverter) FailureToError(f Failure) error {
+	return errors.New(f.Message)
```

Comment:
I think we should have some kind error type here that provides the user access to the failure details. Otherwise, are we expecting Nexus clients to be forced to create their own failure converters just to see "details" on failures?

Reply:
Not following exactly. I understand complex handler side, but I am talking about client side. If I am a simple Nexus caller that wants the error details, how do I get them?

Reply:
What I ended doing in the server code base is to define a `FailureError` struct that's both an `error` and embeds a `Failure` object and a custom failure converter.

I'm just trying to discourage users from using failures directly and would rather hide them from the API until there's strong indication that they should be exposed. `Failure` is a wire level construct and shouldn't be relevant to application code, it's mostly for platforms and proxies like Temporal.

Reply:
It's really the details I'm concerned about. They should not only be for platforms/proxies IMO, same as they are not for gRPC or Temporal or whatever. No strong opinions on how you expose setting/retrieving of details from the error.